### PR TITLE
Frontend/feat: Driver Vehicle rate card

### PR DESCRIPTION
**Pull request by Rohit4997**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-driver/src/Screens/BookingOptionsScreen/View.purs [hunk 1]
```diff
@@ -507,3 +535,72 @@ customTV text' textSize' fontStyle' color' =
...
+  , onClick push $ const $ RateCardBannerClick
+  , visibility $ MP.boolToVisibility rateCardLoaded
+  ][  relativeLayout
+      [ width WRAP_CONTENT
+      , height WRAP_CONTENT
+      ][  textView $
+          [ width WRAP_CONTENT
+          , height WRAP_CONTENT
+          , color Color.white900
+          , background if peakTime then Color.green900 else Color.blue800
```
Comment (aryan-13) [inline Frontend/ui-driver/src/Screens/BookingOptionsScreen/View.purs:569]: Use let and put all the props in an object and access it here

File: Frontend/ui-driver/src/Screens/RateCardScreen/View.purs [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+              , color Color.black700
+              ]
+            <> FontStyle.body3 CT.TypoGraphy
+          , linearLayout
+            [ height WRAP_CONTENT
+            , width WRAP_CONTENT
+            , gravity CENTER
+            , margin $ MarginTop 4
+            , gravity CENTER 
+            , onClick push $ const $ OpenLink "https://www.youtube.com/shorts/NUTNKPzslpw"
```
Comment (aryan-13) [inline Frontend/ui-driver/src/Screens/RateCardScreen/View.purs:379]: Is this supposed to be hardcoded?

**Review by aryan-13 — DISMISSED** [review]

File: Frontend/ui-driver/src/Flow.purs [hunk 3]
```diff
@@ -1693,10 +1697,21 @@ bookingOptionsFlow = do
...
+          farePolicyHour : Nothing
         }
       )
 
+rateCardScreenFlow :: FlowBT String Unit
+rateCardScreenFlow = do
+  action <- UI.rateCardScreen
+  case action of
+    TA.REFRESH_RATE_CARD _ -> do
+      modifyScreenState $ RateCardScreenStateType (\rateCardScreen -> rateCardScreen{props{ reRenderSlider = true }})
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Flow.purs:1712]: @Rohit4997 do we need to exit the screen to rerender slider? Can we remove this if not required.

File: Frontend/ui-driver/src/Screens/BookingOptionsScreen/Controller.purs [hunk 4]
```diff
@@ -101,3 +152,39 @@ dummyVehicleP =
   { vehicleName: ""
   , isSelected: false
   }
+
+getPerKmRate :: API.ServiceTierType -> API.GetDriverRateCardRes -> Maybe Number
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/BookingOptionsScreen/Controller.purs:156]: @Rohit4997 can we combine these three function.

File: Frontend/ui-driver/src/Screens/BookingOptionsScreen/Controller.purs [hunk 5]
```diff
@@ -101,3 +152,39 @@ dummyVehicleP =
...
+getRateCardData :: API.ServiceTierType -> API.GetDriverRateCardRes -> Maybe CTA.BreakupList
+getRateCardData serviceTierType (API.GetDriverRateCardRes rateCardResp) = do
+  let rateCardRespItem = find (\(API.RateCardRespItem item) -> item.serviceTierType == serviceTierType) rateCardResp
+      tips = fromMaybe 0 $ DF.maximum $ tipConfigData (getValueToLocalStore DRIVER_LOCATION) $ getVehicleMapping serviceTierType
+  case rateCardRespItem of
+    Just (API.RateCardRespItem rateCardRespItem) -> Just $ getFareBreakupList rateCardRespItem.rateCardItems tips
+    Nothing -> Nothing
+
+getVehicleMapping :: API.ServiceTierType -> String
+getVehicleMapping serviceTierType = case serviceTierType of
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/BookingOptionsScreen/Controller.purs:179]: move this function to utils.

File: Frontend/ui-driver/src/Screens/RateCardScreen/Controller.purs [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    | ShowRateCard Int
+    | PrimaryButtonAC PrimaryButton.Action
+    | SliderCallback Int
+    | AfterRender
+    | RateCardAction RateCard.Action
+    | ChangeSlider Boolean
+    | OpenLink String
+
+instance showAction :: Show Action where
+    show _ = ""
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/RateCardScreen/Controller.purs:44]: add logs for this screen.

File: Frontend/ui-driver/src/Screens/RateCardScreen/View.purs [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+import Data.Int as DI
+import Data.Array as DA
+import Debug
+
+screen :: ST.RateCardScreenState -> Screen Action ST.RateCardScreenState ScreenOutput
+screen initialState =
+    { initialState
+    , view: view
+    , name: "RateCardScreen"
+    , globalEvents: [ (\push -> pure $ pure unit) ]
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/RateCardScreen/View.purs:57]: assign `[]` to global events

**Review by MercyQueen — CHANGES_REQUESTED** [review]

File: Frontend/ui-driver/src/Screens/RateCardScreen/View.purs (hunk 2 above)
Comment (Rohit4997) [reply Frontend/ui-driver/src/Screens/RateCardScreen/View.purs:378]: Yes, We can hardcode

File: Frontend/ui-driver/src/Screens/BookingOptionsScreen/Controller.purs [hunk 8]
```diff
@@ -77,6 +89,47 @@ eval CallSupport state = do
...
+            fareInfoDescription = rateCardData.fareInfo,
+            isNightShift = rateCardData.isNightShift,
+            nightChargeTill = rateCardData.nightChargeEnd,
+            nightChargeFrom = rateCardData.nightChargeStart,
+            extraFare = rateCardData.fareList
+          }} , props {showRateCard = true } }
+      Nothing -> continue state
+  Nothing -> continue state
+
+eval (RateCardAction RateCard.Close) state = continue state { props { showRateCard = false } , data{rateCard{onFirstPage = false,currentRateCardType = CTA.DefaultRateCard}}}
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/BookingOptionsScreen/Controller.purs:119]: create a common function for Close and BackPressed.

**Review by MercyQueen — APPROVED** [review]

**Review by ashishsingh101 — APPROVED** [review]

**Pull request merged**